includes/device-groups.inc.php

```diff
@@ -153,6 +152,11 @@ function GenGroupSQL($pattern, $search = '', $extra = 0)
     if (!empty($join)) {
         $join = '('.rtrim($join, '& ').') &&';
     }
+  //patch sql error adding extra devices table in query when it has 3 elements 
+    $pos=array_search('devices',$tables);
```

Comment:
This line is unused.

It would possibly be better to just call `array_unique($tables)` to remove any duplicate tables.